### PR TITLE
fix(generators): fix small issues with generators

### PR DESCRIPTION
**Pull request by Loonride**
**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
Not yet

**If relevant, did you update the documentation?**
N/A

**Summary**

- made `sw.js` use correct template path for init generator
- made mkdirp sync, it seems problematic that it was async before since we never know when the folder is created
- removed use of this.async() for the yeoman generator, I think this is an old thing that is not needed anymore with new yeoman version
- make generator choose between yarn and npm

Edit:
- more `init-generator` tests
- `addon-generator` tests
- created `entryOption` property for `init-generator` because `webpackOptions.entry` is not always set in the init generator if the user wants the default entry value, so the value of that variable cannot be relied upon later on in the generator
- added rimraf dependency globally (it seems it was supposed to be there because a script uses it)

**Does this PR introduce a breaking change?**
Yes, some slight breaking changes

**Other information**

### CODE REVIEW

File: packages/generators/src/addon-generator.ts [hunk 1]
```diff
@@ -48,10 +49,12 @@ const addonGenerator = (
                 this.log(`
 				Your project must be inside a folder named ${this.props.name}
 				I will create this folder for you.
-				`);
-                mkdirp(this.props.name).catch((err: object): void => {
-                    if (err) console.error('Failed to create directory', err);
-                });
+                `);
+                try {
+                    mkdirp.sync(this.props.name);
```
Comment (jamesgeorge007) [inline packages/generators/src/addon-generator.ts]: Yeah, the `sync` version would be a best fit here.

File: packages/generators/src/addon-generator.ts [hunk 2]
```diff
@@ -69,9 +72,13 @@ const addonGenerator = (
...
-            this.npmInstall(['webpack-defaults', 'bluebird'], {
-                'save-dev': true,
-            });
+            const packager = getPackageManager();
+            const opts: {
+                dev?: boolean;
+                "save-dev"?: boolean;
+            } = packager === "yarn" ? { dev: true } : { "save-dev": true };
+
+		    this.scheduleInstallTask(packager, ['webpack-defaults', 'bluebird'], opts);
```
[the remark this answers is not in the document]
Comment (Loonride) [reply packages/generators/src/addon-generator.ts]: Yes, I will get to adding some tests for `addon-generator` soon.

[the remark this answers is not in the document]
Comment (anshumanv) [reply packages/generators/src/addon-generator.ts]: Thank You! This looks great. ✨

[the remark this answers is not in the document]
Comment (Loonride) [reply packages/generators/src/addon-generator.ts]: Added `addon-generator` tests for this. Currently just testing that the correct arguments are passed in whether it is for yarn or npm. I don't think there is a good way to actually test that the npm install or yarn add is done correctly, as doing an installation in a test will probably just create terrible test instability, but this is something that can be looked at later.

**Comment by Loonride** [comment]
Seems to be a bad test run

**Comment by evilebottnawi** [comment]
@Loonride ignore it, we migrate on github actions

**Pull request merged**